fil_fungible_token/src/token/mod.rs

```diff
@@ -215,10 +219,13 @@ where
             state.attempt_use_allowance(&self.bs, spender, owner, &value)?;
         }
         // attempt to burn the requested amount
-        let new_amount = state.change_balance_by(&self.bs, owner, &value.neg())?;
+        let new_amount = state.change_balance_by(&self.bs, owner, &value.clone().neg())?;
+
+        // decrease total_supply
+        state.change_supply_by(&value.neg())?;
```

Comment:
Nit: extract `value.neg()` to compute negation only once. I'm also curious why a clone was required on 222.